build/ArtifactBuilder/Artifacts/MsiInstaller.cs

```diff
@@ -289,14 +287,163 @@ private static IReadOnlyCollection<string> CreateReadOnlyCollection(params strin
             return new HashSet<string>(items);
         }
 
+        private bool TryGetMsiPath(out string msiPath)
+        {
+            var fileSearchPattern = $@"NewRelicAgent_{Platform}_*.msi";
+            msiPath = Directory.GetFiles(MsiDirectory, fileSearchPattern).FirstOrDefault();
+
+            if (string.IsNullOrEmpty(msiPath))
+            {
+                Console.WriteLine("Warning: The {0} installer could not be found.", fileSearchPattern);
+                return false;
+            }
+
+            return true;
+        }
+
         protected override void ValidateContent()
         {
-            throw new NotImplementedException();
+            var unpackedLocation = Unpack();
+
+            if (string.IsNullOrEmpty(unpackedLocation))
+            {
+                return;
+            }
+
+            var installedFilesRoot = unpackedLocation + $@"\New Relic\.NET Agent\";
+
+            var expectedComponents = GetExpectedComponents(installedFilesRoot);
+
+            var unpackedComponents = GetUnpackedComponents(installedFilesRoot);
+
+            var missingExpectedComponents = new SortedSet<string>(expectedComponents, StringComparer.OrdinalIgnoreCase);
+            missingExpectedComponents.ExceptWith(unpackedComponents);
+            foreach (var missingComponent in missingExpectedComponents)
```

Comment:
Maybe it would be better to collect all the errors and then report? So we don't have to fix-build-fix-etc

Reply:
I thought about that, but for simplicity I kept it following the same behavior as the other similar errors. When we add instrumentation and forget to add it to the MSI we'll see an error that lets us know we missed that adding that instrumentation. Technically there would be 4 possible errors generated, but the first one is good enough for letting us know what the problem is. If we're working on a bigger msi change, we probably need to run the artifact builder locally and treat it as the test project for the msi.